js/style/style.js

```diff
@@ -247,19 +251,95 @@ Style.prototype = util.inherit(Evented, {
         zh.lastZoom = z;
     },
 
+    _checkLoaded: function () {
+        if (!this._loaded) {
+            throw new Error('Style is not done loading');
+        }
+    },
+
     /**
-     * Apply multiple style mutations in a batch
-     * @param {function} work Function which accepts the StyleBatch interface
+     * Apply queued style mutations in a batch
      * @private
      */
-    batch: function(work) {
-        styleBatch(this, work);
+    update: function(classes, options) {
```

Comment:
Yes, great point! I like all the suggested renames except the `change` event, which we use throughout the code for other classes, and it's generally more common to have events like this called `change` rather than `update` in JS APIs.